### ISSUE TITLE
ClassValueMap NPE

### ISSUE THREAD

**Issue by galderz** [graalvm/mandrel#22]
With jdk11u-dev sometimes you see this when building native images:

```
Caused by: java.lang.NullPointerException
        at java.base/java.lang.ClassValue$ClassValueMap.loadFromCache(ClassValue.java:535)
        at java.base/java.lang.ClassValue$ClassValueMap.probeHomeLocation(ClassValue.java:541)
        at java.base/java.lang.ClassValue.get(ClassValue.java:101)
```

This is a bug in the underlying JDK. I'm working on a reproducer using byteman.

This would not happen with Labs JDK because they don't reflection to extract this info, but query it directly from the JVM.

[Longer output](https://gist.github.com/galderz/4426cc2285588286710331b4ddeca1ee).

See https://github.com/quarkusio/quarkus/issues/8776 for first encounter.

**Comment by zakkak** [comment]
@galderz should we mark this as `Not-Stale`?

**Comment by galderz** [comment]
@zakkak I think so. A fix for this is close to making it to jdk, see [here](https://github.com/openjdk/jdk/pull/94).

**Comment by Karm** [comment]
Hit it again earlier today: https://ci.modcluster.io/job/mandrel-20.1-linux-test/OS_AXIS=centos8%7C%7Crhel8/135/consoleText

Quarkus 1.7.5, Mandrel 20.1 branch.

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.smallrye.graphql.GreetingsResourceTest
2020-10-06 01:49:47,668 INFO  [io.quarkus] (main) Quarkus 1.7.5.Final on JVM started in 1.763s. Listening on: http://0.0.0.0:8081
2020-10-06 01:49:47,669 INFO  [io.quarkus] (main) Profile test activated. 
2020-10-06 01:49:47,670 INFO  [io.quarkus] (main) Installed features: [cdi, smallrye-graphql]
ANTLR Tool version 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.8ANTLR Tool version 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.8[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.914 s - in io.quarkus.it.smallrye.graphql.GreetingsResourceTest
[INFO] Running io.quarkus.it.smallrye.graphql.ApplicationPropertiesOverrideTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.029 s - in io.quarkus.it.smallrye.graphql.ApplicationPropertiesOverrideTest
[INFO] Running io.quarkus.it.smallrye.graphql.GreetingResourceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.037 s - in io.quarkus.it.smallrye.graphql.GreetingResourceTest
2020-10-06 01:49:49,335 INFO  [io.quarkus] (main) Quarkus stopped in 0.029s
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ quarkus-integration-test-smallrye-graphql ---
[INFO] 
[INFO] --- maven-source-plugin:3.0.1:jar-no-fork (attach-sources) @ quarkus-integration-test-smallrye-graphql ---
[INFO] Building jar: /home/jenkins/jenkins/workspace/mandrel-20.1-linux-test/OS_AXIS/centos8||rhel8/quarkus/integration-tests/smallrye-graphql/target/quarkus-integration-test-smallrye-graphql-1.7.5.Final-sources.jar
[INFO] 
[INFO] --- quarkus-maven-plugin:1.7.5.Final:build (default) @ quarkus-integration-test-smallrye-graphql ---
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: /home/jenkins/jenkins/workspace/mandrel-20.1-linux-test/OS_AXIS/centos8||rhel8/quarkus/integration-tests/smallrye-graphql/target/quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner.jar
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 493ms
[INFO] 
[INFO] --- quarkus-maven-plugin:1.7.5.Final:native-image (native-image) @ quarkus-integration-test-smallrye-graphql ---
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /home/jenkins/jenkins/workspace/mandrel-20.1-linux-test/OS_AXIS/centos8||rhel8/quarkus/integration-tests/smallrye-graphql/target/quarkus-integration-test-smallrye-graphql-1.7.5.Final-native-image-source-jar/quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /home/jenkins/jenkins/workspace/mandrel-20.1-linux-test/OS_AXIS/centos8||rhel8/quarkus/integration-tests/smallrye-graphql/target/quarkus-integration-test-smallrye-graphql-1.7.5.Final-native-image-source-jar/quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on Snapshot
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /home/jenkins/jenkins/workspace/mandrel-20.1-linux-test/OS_AXIS/centos8||rhel8/mandrel-java11-20.1-SNAPSHOT/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -jar quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http -H:NativeLinkerOption=-no-pie -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner
[quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner:23191]    classlist:   3,435.73 ms,  0.96 GB
[quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner:23191]        (cap):     631.90 ms,  0.96 GB
[quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner:23191]        setup:   2,256.44 ms,  0.96 GB
01:50:13,491 INFO  [org.jbo.threads] JBoss Threads version 3.1.1.Final
ANTLR Tool version 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.8ANTLR Tool version 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.8[quarkus-integration-test-smallrye-graphql-1.7.5.Final-runner:23191]     analysis:  22,194.84 ms,  3.28 GB
5 fatal errors detected:
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
  at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
  at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
  at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
  at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
  at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
  at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
  at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
  at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
  at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
  ... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
  at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  ... 26 more
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
  at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
  at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
  at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
  at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
  at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
  at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
  at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
  at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
  at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
  ... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
  at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  ... 26 more
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
  at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
  at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
  at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
  at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
  at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
  at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
  at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
  at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
  at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
  ... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
  at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  ... 26 more
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
  at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
  at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
  at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
  at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
  at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
  at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
  at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
  at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
  at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
  ... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
  at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  ... 26 more
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
  at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
  at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
  at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
  at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
  at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
  at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
  at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
  at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
  at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
  at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
  at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
  at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
  ... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
  at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  ... 26 more
```

**Comment by galderz** [comment]
@Karm What did you hit again? This issue is about an NPE...

**Comment by galderz** [comment]
Ah, I see the exception in the original bug. The issue has been fixed in jdk/jdk and its on its way to being backported to jdk/11.  FYI [JDK-8251397](https://bugs.openjdk.java.net/browse/JDK-8251397).

**Comment by galderz** [comment]
Actually, it's already part of jdk11u-dev (see [JDK-8254025](https://bugs.openjdk.java.net/browse/JDK-8254025)), so should be fixed in JDK 11.0.10.

**Comment by zakkak** [comment]
Mandrel is now based on 11.0.10 and AFAIK the issue has not appeared again.
